.github/workflows/draft_new_release.yml

```diff
@@ -9,32 +9,72 @@ jobs:
     name: "Draft a new release"
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f
-        with:
-          ref: 'release'
       - name: Extract version from milestone
         run: |
           VERSION="${{ github.event.milestone.title }}"
-          echo "RELEASE_VERSION=${VERSION/v/}" >> $GITHUB_ENV
+          RELEASE_VERSION="${VERSION/v/}"
+          # Transforms 1.13.2 to 1.13 so that we can re-use the same
+          # branch for patch releases.
+          BRANCH_VERSION="${RELEASE_VERSION:$i:-2}"
+          if [[ "${RELEASE_VERSION: -1}" == "0" ]]; then
+            CHECKOUT_REF="develop"
+          else
+            CHECKOUT_REF="release-${BRANCH_VERSION}"
+          fi
+
+          # Export variables separately so the scripting above is more legible,
+          # and we can actually use them within this block. Changes to $GITHUB_ENV
+          # only affect the next step, not the current one.
```

Comment:
`$GITHUB_ENV` note is here to document what I learned today, no strong preference to keeping or removing it.